github-upload-public-key/main.tf

```diff
@@ -0,0 +1,55 @@
+terraform {
+  required_version = ">= 1.0"
+
+  required_providers {
+    coder = {
+      source  = "coder/coder"
+      version = ">= 0.12"
+    }
+  }
+}
+
+variable "agent_id" {
+  type        = string
+  description = "The ID of a Coder agent."
+}
+
+variable "external_auth_id" {
+  type        = string
+  description = "The ID of the GitHub external auth."
+  default     = "github"
+}
+
+variable "github_api_url" {
+  type        = string
+  description = "The URL of the GitHub instance."
+  default     = "https://api.github.com"
+}
+
+// Optional variables mostly for testing purposes, will normally come from data.coder_workspace.me
+variable "access_url" {
+  type        = string
+  description = "The access URL of the workspace."
+  default     = ""
+}
+
+variable "owner_session_token" {
+  type        = string
+  description = "The owner session token of the workspace."
+  default     = ""
+}
```

Comment:
You should be able to pass these as env vars when running the script instead, that way we don't have to have these extra vars.

Reply:
Yeah I can't think of. Use case where thet will be different from the provider provided values.

Reply:
I was able to do this 👍